### ISSUE TITLE
Design teleportation as an action effect

### ISSUE THREAD

**Issue by ericyc16** [uchicago-cs/chiventure#550]
While there is currently support for several different types of effects, one effect that is not yet supported is teleportation. Issue #239 provides an example of what this effect could do. The basic idea is that if an item with this effect is interacted with, then the player is transported to a location. Work on the design for such a feature, determining details such as how this can be implemented and where in the code it should go.

**Comment by ericyc16** [comment]
Created a wiki for the design, which can be found [here](https://github.com/uchicago-cs/chiventure/wiki/Action-Management-Teleportation-Effect-Design). As of writing this, though, there is very little on the page. Next step is to work on the design.

**Comment by ericyc16** [comment]
Right now, I'm leaning towards the idea of making teleportation its own action, but making this action only supportable as an effect. This is because teleportation would essentially change the game's `curr_room`, which is similar to what path actions already do. However, players should not normally be able to teleport. Implementing teleportation as an action would also potentially be useful in the future if some other new functionality is provided into chiventure, such as spells.

I've talked to @isaacfink21, who is currently working on overhauling the `game_action_effect_t` struct. Even though effects currently can only be applied to items, it sounds like it could be possible to allow teleportation to be an effect.

Another thing I might want to consider is utilizing the functionality that the custom actions team is working on implementing, since teleportation as an effect would be quite complex. 

For now, I will consider working on the design while also exploring code to determine what solutions are possible.

**Comment by ericyc16** [comment]
I have decided that it makes the most sense to implement teleportation as an action that cannot normally be executed. This action will instead be used only under special circumstances, such as as the effect of completing a different action or as a spell unlocked via skill trees. Note that [the wiki](https://github.com/uchicago-cs/chiventure/wiki/Action-Management-Teleportation-Effect-Design) has been updated with more details to reflect this decision.

However, I do not believe that currently, this effect can be used in the chiventure code. While it could be useful later down the line when more progress has been made to this project, it will not be useful now. Therefore, I am questioning whether or not it makes sense to implement this design now.

I will ask my team members and TA their opinions on this and make a decision later.

**Comment by ericyc16** [comment]
After talking to @MaxineK36 and @isaacfink21, I have decided that it does not make sense to implement teleportation as a pre-defined action.

If I were to implement teleportation as a pre-defined action defined by action management, it would be an action that can only be executed under special circumstances. Therefore, I think that it makes more sense for it to instead be defined by a game author via a custom action. Based on [the design documentation provided by the custom actions team](https://github.com/uchicago-cs/chiventure/wiki/Custom-Actions-Design-Document#move-player-room), it seems like they are already planning on introducing moving a character to a room as an atomic effect.

Because I have decided not to implement teleportation as a pre-defined action, I am now closing this issue.
